common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -146,7 +146,7 @@ public void pushBlocks(
               assert inputListener instanceof BlockPushingListener :
                 "Expecting a BlockPushingListener, but got " + inputListener.getClass();
               TransportClient client = clientFactory.createClient(host, port);
-              new OneForOneBlockPusher(client, appId, transportConf.appAttemptId(), inputBlockId,
+              new OneForOneBlockPusher(client, appId, appAttemptId, inputBlockId,
```

Comment:
use `getAppAttemptId`

Reply:
Changed to use comparableAppAttemptId, and since it is private var, so I don't choose to use the getAppAttemptId() here.